pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTExplicitConstructorInvocation.java

```diff
@@ -96,4 +100,15 @@ public ASTTypeArguments getExplicitTypeArguments() {
     public ASTExpression getQualifier() {
         return AstImplUtil.getChildAs(this, 0, ASTExpression.class);
     }
+
+    @Override
+    public OverloadSelectionResult getOverloadSelectionInfo() {
+        getTypeMirror(); // force evaluation
```

Comment:
I think, we should be careful to add such side effects.... getTypeMirror() sounds innocent, but obviously it calls back setOverload... so it really does resolve types+overloads and then returns the resolved type... that's probably a result of doing lazy typeres, isn't it?

btw: this method throw a NPE, if no overloadselectionresult could be resolved, but the previous class returns null (with an assert check). Is this intended?

Reply:
> getTypeMirror() sounds innocent, but obviously it calls back setOverload... so it really does resolve types+overloads and then returns the resolved type... that's probably a result of doing lazy typeres, isn't it?

Yes, the only entry point to trigger lazy typeres is getTypeMirror. For invocation nodes, this will start an overload resolution, as in the general case, this is the only way to get the return type of a method for example. The "good" thing about using getTypeMirror to force the evaluation, is that it is already guarded and will execute at most once. We could have another method `forceTypeRes`, just to make the intention clearer.. 

> btw: this method throw a NPE, if no overloadselectionresult could be resolved, but the previous class returns null (with an assert check). Is this intended?

Normally even if no overload can be found, setOverload will be set anyway with a failed (but non-null) result. So it would be a bug if it was null after overload resolution. We can probably be more consistent with what exceptions we use (whether NPE or assert)... 

Here is the logic that sets the overload selection info (where NO_CTDECL is the sentinel for a failed overload selection):

https://github.com/pmd/pmd/pull/2689/files#diff-d86db01bfc649e5de55423a6d27f09c1R165